### DISCUSSION TITLE
[custom wallet] update to 2.x, how to add a connector for custom wallet & how to use getUri when is an async function

### DISCUSSION THREAD

**Discussion by fedCat2024** [rainbow-me/rainbowkit#2012] [Q&A]
when i update to 2.x, i found that:
1. how to add a connector for a custom wallet
in 1.x it has a connector attribute, as below
<img width="825" alt="image" src="https://github.com/rainbow-me/rainbowkit/assets/163102407/10114789-aeb9-416b-81b4-8e06deda514e">

2. getUri  is now sync function.
in 1.x is a promise, as below
<img width="860" alt="image" src="https://github.com/rainbow-me/rainbowkit/assets/163102407/82463137-b796-47ca-ada8-0d330e5b107a">

this is my 1.x code:

<img width="894" alt="image" src="https://github.com/rainbow-me/rainbowkit/assets/163102407/596508ac-78d7-4c28-b11f-c5a54478bce9">

how to update to 2.x?

**Comment by 0xmkh** [comment]
@fedCat2024 Here is a [safeWallet ](https://github.com/rainbow-me/rainbowkit/blob/main/packages/rainbowkit/src/wallets/walletConnectors/safeWallet/safeWallet.ts) example. You can also look at our docs [here](https://www.rainbowkit.com/docs/custom-wallets#examples). Let me know if this helps 👍
